lib/dnsimple/client.rb

```diff
@@ -147,16 +151,11 @@ def execute(method, path, data, options = {})
     #
     # @param  [String] method The HTTP method
     # @param  [String] path The path, relative to {#base_url}
-    # @param  [Hash] options Query and header params for request
+    # @param  [Hash] data The body for the request
+    # @param  [Hash] options The query and header params for the request
     # @return [HTTParty::Response]
     def request(method, path, data, options = {})
-      if data.is_a?(Hash)
-        options[:query]   = data.delete(:query)   if data.key?(:query)
-        options[:headers] = data.delete(:headers) if data.key?(:headers)
-      end
-      if !data.empty?
-        options[:body] = data
-      end
+      options[:body] = data if !data.empty?
```

Comment:
FYI, I believe there is currently a bug where we don't serialize the payload as JSON. This specific line should probably be

    options[:body] = data.to_json

if the request type is JSON. I also wonder if the `!data.empty?` is necessary as when `data` is nil it will simply call

    options[:body] = nil